pod.go

```diff
@@ -179,6 +201,31 @@ func (pc *podCollector) Collect(ch chan<- prometheus.Metric) {
 	}
 }
 
+func kubeLabelsToPrometheusLabels(labels map[string]string) ([]string, []string) {
+	labelKeys := make([]string, len(labels))
+	labelValues := make([]string, len(labels))
+	i := 0
+	for k, v := range labels {
+		labelKeys[i] = "label_" + sanitizeLabelName(k)
+		labelValues[i] = v
+		i++
+	}
+	return labelKeys, labelValues
+}
+
+func sanitizeLabelName(s string) string {
+	return invalidLabelCharRE.ReplaceAllString(s, "_")
+}
+
+func podLabelsDesc(labelKeys []string) *prometheus.Desc {
+	return prometheus.NewDesc(
```

Comment:
This is actually just a validation that the go client library performs that we turned off. Metrics/time-series are _only_ identified by their unique label-set. The description registering was a premature validation attempt that is too hard of a validation in cases like this, although typically a correctly validation. So tl;dr yes you're right.